stripe.go

```diff
@@ -101,12 +104,53 @@ type Backend interface {
 	SetMaxNetworkRetries(maxNetworkRetries int)
 }
 
-// BackendConfiguration is the internal implementation for making HTTP calls to Stripe.
+// BackendConfig is used to configure a new Stripe backend.
+type BackendConfig struct {
+	// HTTPClient is an HTTP client instance to use when maxing requests.
+	//
+	// If left unset, it'll be set to a default HTTP client for the package.
+	HTTPClient *http.Client
+
+	// LogLevel is the logging level of the library and defined by:
+	//
+	// 0: no logging
+	// 1: errors only
+	// 2: errors + informational (default)
+	// 3: errors + informational + debug
+	//
+	// Defaults to 0 (no logging), so please make sure to set this if you want
+	// to see logging output in your custom configuration.
+	LogLevel int
```

Comment:
I don't think this is ever used; should it have a value in `BackendConfiguration` and be used there too? Given that it will default to zero it may not be possible to determine if we should use this or the global LogLevel, I'm not sure how to do that in a backwards compatible way.

Reply:
> I don't think this is ever used; should it have a value in BackendConfiguration and be used there too?

Haha, that's what I get for trying to push through a change like this near the end of the day!

Yep, that's a major omission. I just added `LogLevel` to `BackendConfiguration` and changed all existing checks on `LogLevel` to use it.